backend/src/Core/Availability/Parse.hs

```diff
@@ -0,0 +1,105 @@
+module Core.Availability.Parse (parseOccupancies, icalendarToOccupancies) where
+
+import Data.ByteString.Lazy (ByteString)
+import Data.Tuple (swap)
+import Data.Default (def)
+import Data.Map.Internal (elems)
+import Text.ICalendar
+import Data.Time
+import Data.Maybe (mapMaybe)
+import Data.Text.Lazy (unpack)
+
+import Utils.Datatypes (UserId, UserOccupancies (..), Occupancy (Occupancy), TimeRange (..))
+
+type ImportError = String
+type ImportWarning = String
+
+-- | First parses a @ByteString@ into a @VCalendar@, and then interprets the @VCalendar@ into
+-- @UserOccupancies@. It does so by calling @icalendarToOccupancies@ and combining the resulting
+-- @Occupancy@ values with the user id, creating @UserOccupancies@.
+parseOccupancies :: UserId -> ByteString -> Either ImportError ([ImportWarning], UserOccupancies)
+parseOccupancies user bytes = fmap (UserOccupancies user . concatMap icalendarToOccupancies) . swap
+                             <$> parseICalendar def "Uploaded icalendar" bytes
+
+-- | Grabs all occupancies as described by an iCalendar.
+-- Note: events which we failed to fully parse as an occupancy are not included
+-- Note: as explained in @grabOccupancies@, one event might be represented by multiple occupancies.
+icalendarToOccupancies :: VCalendar -> [Occupancy]
+icalendarToOccupancies cal = let events = elems $ vcEvents cal
+                      in concat $ mapMaybe grabOccupancies events
+
+-- | Interprets a @VEvent@ into multiple @Occupancy@ values.
+-- It is in the definition of a @Occupancy@ that one is bound to a single day.
+-- We thus might find multiple occupancies here (one for each day).
+grabOccupancies :: VEvent -> Maybe [Occupancy]
+grabOccupancies e = do
+  title <- grabTitle e
+  occupancies <- grabOccupancyTimeRanges e
+  return $ map (\(TimeRange date start end) -> Occupancy title date start end) occupancies
+
+-- | Grabs a title from a @VEvent@, if present.
+grabTitle :: VEvent -> Maybe String
+grabTitle e = unpack . summaryValue <$> veSummary e
+
+-- | Determines the raw timewise occupancy described by the given @VEvent@, and returns it in the shape of concurrent @TimeRange@ values.
+-- It is in the definition of a @TimeRange@ that one is bound to a single day. We thus might need to return multiple ranges here.
+grabOccupancyTimeRanges :: VEvent -> Maybe [TimeRange]
+grabOccupancyTimeRanges e = do
+  utcStart <- dateTimeToUTC . dtStartDateTimeValue <$> veDTStart e
+  utcEnd   <- dateTimeToUTC . dtEndDateTimeValue <$> fmap (fromDTEndDurationToDTEnd utcStart) (veDTEndDuration e)
+  return $ toTimeRanges utcStart utcEnd
+
+-- FIXME this function assumes timezones if not present, can be determined by global timezone calendar element.
```

Comment:
There is a ```FIXME```here. Can that be removed?

Reply:
Well we can spend time on fully supporting timezones, but that's a waste of time. Removing the FIXME removes information about the bug. In name of transparancy I believe it should be kept, and others can understand why we have not fixed it

Reply:
Perhaps we can rename it to TODO